lasagne/nonlinearities.py

```diff
@@ -286,6 +286,71 @@ def softplus(x):
     return theano.tensor.nnet.softplus(x)
 
 
+# Clipped Activation
+class ClippedActivation(object):
+    """Clipped Activation activation function
+    :math:`\\varphi(x,c,a) = \\min(a(x),c)`
+
+    This function clips the activation function. Usefull for non-saturating
+    functions such as the rectify to avoid exploding activations.
+
+    Parameters
+    ----------
+    x : float32
+        The activation (the summed, weighted input of a neuron).
+    clip : float32
+        The upper bound for the activation function.
+    activation : function
+        An activation function that maps x onto a float32
+
+    Methods
+    -------
+    __call__(x)
+        Apply the clipping to the to the activation `x`.
+
+    Examples
+    --------
+    In contrast to other activation functions in this module, this is
+    a class that needs to be instantiated to obtain a callable:
+
+    >>> from lasagne.layers import InputLayer, DenseLayer
+    >>> l_in = InputLayer((None, 100))
+    >>> from lasagne.nonlinearities import ClippedActivation, rectify
+    >>> custom_activation = ClippedActivation(clip=20, activation=rectify)
+    >>> l1 = DenseLayer(l_in, num_units=200, nonlinearity=custom_activation)
+
+    Alternatively, you can use the provided instance for clip=20,
+    activation=rectify
+
+    >>> from lasagne.nonlinearities import clipped_rectify
+    >>> l2 = DenseLayer(l_in, num_units=200, nonlinearity=clipped_rectify)
+    """
+
+    def __init__(self, clip=20, activation=rectify):
+        self.clip = clip
+        self.activation = activation
+
+    def __call__(self, x):
+        return theano.tensor.minimum(self.activation(x), self.clip)
+
+
+# shortcut with default activation and clip
+clipped_rectify = ClippedActivation()
+clipped_rectify.__doc__ = """clipped_rectify(x)
+
+    Instance of :class:`ClippedActivation' such as used in [1]_.
+    :math:`\\clip=20.0 \\activation=rectify`
+
+    References
+    ----------
+    .. [1] Awni Hannun, Carl Case, Jared Casper, Bryan Catanzaro, Greg Diamos,
+            Eric Elsen, Ryan Prenger, Sanjeev Satheesh, Shubho Sengupta,
+            Adam Coates, Andrew Y. Ng (2014):
+       Deep Speech: Scaling up end-to-end speech recognition,
+       http://arxiv.org/abs/1412.5567
```

Comment:
Sphinx can be very picky about indentations -- if it doesn't render correctly, you may have to indent everything the same.